wamr-compiler/main.c

```diff
@@ -231,6 +231,7 @@ main(int argc, char *argv[])
             option.enable_bulk_memory = true;
             option.enable_thread_mgr = true;
             option.enable_ref_types = false;
+            option.enable_aux_stack_check = false;
```

Comment:
Should not disable aux stack check for multi-thread, it is enabled for lib-pthread mode, which divides aux stack area into many regions and assign each region to a thread, the bottom and boundary of a region are pre-known and are set to thread exec_env's aux_stack_bottom and aux_stack_boundary fields.

The wasi-threads mode allocates the aux stack from the libc heap of wasm app, the bottom and boundary of the the aux stack are also known when the aux stack is allocated (seems bottom = addr_allocated + size, boundary = addr_allocated), I think we should set them to exec_env's aux_stack_bottom and aux_stack_boundary, but not change code here.

Reply:
For wasi-threads the aux stack boundaries are not known for the runtime though. There were some discussions here https://github.com/WebAssembly/wasi-threads/issues/12 and here https://github.com/WebAssembly/wasi-libc/pull/380; there's no final conclusion although I'm leaning towards having this part of C ABI and not runtime ABI, so all the stack check work will most likely be done in the user space code.

Reply:
If so, could you set `exec_env->aux_stack_bottom` to UINT32_MAX and `exec_env->aux_stack_boundary` to 0? So the aux stack boundary checks will always succeed and no need to change code here.

Reply:
Right, I tried it and it works with AOT.
I kept `wasm_exec_env_is_aux_stack_managed_by_runtime()` to use it in `free_aux_stack()`.